### PR TITLE
feat: blockchain sync

### PR DESCRIPTION
**Pull request by tomaszantas**
Description
---

- [x] Add `ProgressBar` component
- [x] Add Blockchain sync step to the onboarding that start the BaseNode

In next PRs:

- listen to the backend events about Blockchain sync status, and replace mocked `progress` and `remaining time` values

Motivation and Context
---

#319 (partial implementation)

How Has This Been Tested?
---

https://user-images.githubusercontent.com/11715931/175574907-6357ebc4-917a-4340-bf0d-f85f5217481b.mov

### CODE REVIEW

File: applications/launchpad/gui-react/src/components/Onboarding/OnboardingMessages/LastStepsMessages.tsx [hunk 1]
```diff
@@ -1,33 +1,175 @@
...
+    ])
+    try {
+      await dispatch(baseNodeActions.startNode()).unwrap()
+    } catch (e) {
+      setError(true)
+    }
+  }
+
+  useEffect(() => {
+    if (error) {
```
Comment (tarnas14) [inline applications/launchpad/gui-react/src/components/Onboarding/OnboardingMessages/LastStepsMessages.tsx]: it's not necessary to depend on state and react to it in useEffect

make this a function and call it instead of `setError(true)`

File: applications/launchpad/gui-react/src/components/Onboarding/OnboardingMessages/LastStepsMessages.tsx [hunk 2]
```diff
@@ -1,33 +1,175 @@
...
+  }, [error])
+
+  useEffect(() => {
+    if (syncStarted && updateMessageBoxSize && contentRef.current) {
+      updateMessageBoxSize()
+    }
+  }, [syncStarted])
+
+  useEffect(() => {
+    if (syncFinished) {
```
Comment (tarnas14) [inline applications/launchpad/gui-react/src/components/Onboarding/OnboardingMessages/LastStepsMessages.tsx]: same as above -> call this logic directly instead of adding state which fires off an useEffect

File: applications/launchpad/gui-react/src/components/TBot/TBotPrompt/MessageBox.tsx [hunk 3]
```diff
@@ -58,11 +68,17 @@ const MessageBox = (
       <MessageSpaceContainer>
         <MessageSlideIn style={{ ...useSlideInAnim }}>
           <StyledMessage
+            ref={messageBoxRef}
             style={{ ...useOpacityAnim }}
             $skipButton={skipButton}
             $floating={floating}
           >
-            {children}
+            {React.Children.map(children, child => {
```
Comment (tarnas14) [inline applications/launchpad/gui-react/src/components/TBot/TBotPrompt/MessageBox.tsx:76]: why is this necessary? I think you should isolate this logic and add a comment explaining why we react clone, cause this is usually pretty specific

File: applications/launchpad/gui-react/src/locales/onboarding.ts [hunk 4]
```diff
@@ -45,7 +46,10 @@ const translations = {
     },
   },
   lastSteps: {
-    message1: 'Awesome, everything went smoothly. You are one step away from starting mining.'
+    message1: 'Awesome, everything went smoothly. You are one step away from starting mining.',
+    message2: 'When the data synchronization is completed, the Tari Launchpad dashbord will start automatically. Do not close this window as this will pause the entire process.',
```
Comment (tarnas14) [inline applications/launchpad/gui-react/src/locales/onboarding.ts]:
```suggestion
    message2: 'When the data synchronization is completed, the Tari Launchpad dashboard will start automatically. Do not close this window as this will pause the entire process.',
```

**Review by tarnas14 — CHANGES_REQUESTED** [review]

File: applications/launchpad/gui-react/src/components/Onboarding/OnboardingMessages/LastStepsMessages.tsx [hunk 5]
```diff
@@ -1,33 +1,169 @@
...
+        </CtaButtonContainer>
+      )}
+
+      {syncStarted && (
+        <>
+          <Progress progress={progress} time={remainingTime} />
+          <CtaButtonContainer style={{ justifyContent: 'center' }}>
+            <Button
+              variant='secondary'
+              onClick={() => dispatch(setOnboardingComplete(true))}
```
Comment (tarnas14) [inline applications/launchpad/gui-react/src/components/Onboarding/OnboardingMessages/LastStepsMessages.tsx]: how does the application behave in this state?
if the user cancelled during sync or killed containers during sync - do we have any way of catching that state and fixing it?

File: applications/launchpad/gui-react/src/components/TBot/TBotPrompt/MessageBox.tsx [hunk 6]
```diff
@@ -58,11 +68,27 @@ const MessageBox = (
...
+              if (React.isValidElement(child)) {
+                /**
+                 * TBot Prompt adds each message twice to the DOM:
```
Comment (tarnas14) [inline applications/launchpad/gui-react/src/components/TBot/TBotPrompt/MessageBox.tsx:79]: :+1:

File: applications/launchpad/gui-react/src/locales/onboarding.ts [hunk 7]
```diff
@@ -59,7 +60,10 @@ const translations = {
     },
   },
   lastSteps: {
-    message1: 'Awesome, everything went smoothly. You are one step away from starting mining.'
+    message1: 'Awesome, everything went smoothly. You are one step away from starting mining.',
+    message2: 'When the data synchronization is completed, the Tari Launchpad dashboard will start automatically. Do not close this window as this will pause the entire process.',
+    syncError: 'Ups! Something went wrong!',
```
Comment (tarnas14) [inline applications/launchpad/gui-react/src/locales/onboarding.ts]:
```suggestion
    syncError: 'Oops! Something went wrong!',
```

**Review by tarnas14 — DISMISSED** [review]

File: applications/launchpad/gui-react/src/components/Onboarding/OnboardingMessages/LastStepsMessages.tsx (hunk 5 above)
Comment (tomaszantas) [reply applications/launchpad/gui-react/src/components/Onboarding/OnboardingMessages/LastStepsMessages.tsx]: Originally, it was moving the user to the Home screen where Base Node and Tor were running. But on the second thought, I believe it should just close the application and stop containers. I changed this in last commit, so it closes the app. But waiting for the confirmation from Tola.

Comment (tarnas14) [reply applications/launchpad/gui-react/src/components/Onboarding/OnboardingMessages/LastStepsMessages.tsx]: ok, makes sense!

File: applications/launchpad/gui-react/src/components/ProgressBar/index.tsx [hunk 8]
```diff
@@ -0,0 +1,52 @@
+import { useEffect, useRef, useState } from 'react'
+import { config, useSpring } from 'react-spring'
+import Text from '../Text'
+import { Track, Fill, StyledProgressBar, Tip } from './styles'
+
+import { ProgressBarProps } from './types'
+
+const limitValue = (value: number) => {
+  return value > 100 ? 100 : Math.abs(value)
```
Comment (tarnas14) [inline applications/launchpad/gui-react/src/components/ProgressBar/index.tsx:9]: just a tip: you can do things like this with `Math.min(100, Math.abs(value))`

I don't think there's any actual difference in the speed or amount of operations, but to me it looks nicer xD

**Review by tarnas14 — APPROVED** [review]
